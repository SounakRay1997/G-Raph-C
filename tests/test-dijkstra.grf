int main() {

    node a = makeNode("a1", 1);
    node b = makeNode("b1", 2);
    node c = makeNode("c1",3);
    node d = makeNode("d1",5);
    edge e1 = makeEdge(a,b,10,true);
    edge e2 = makeEdge(b,d,10,true);
    edge e3 = makeEdge(a,c,20,true);
    edge e4 = makeEdge(c,d,40,true);
    nlist nl = createNList(); 
    elist el = createEList();
    appendNode(nl,a);
    appendNode(nl,b);
    appendNode(nl,c);
    appendNode(nl,d);
    appendEdge(el,e1);
    appendEdge(el,e2);
    appendEdge(el,e3);
    appendEdge(el,e4);
    
    graph g = makeGraph(nl,el);
    node x = getNodeFromGraph(g,"c1");
    edge E = getEdgeFromGraph(g,b,c);
    
    nlist l = createNList();
    l=getNeighbours(g,a);
    print(nlistLen(l));
    djikstras(g,a,d);

    return 0;
}