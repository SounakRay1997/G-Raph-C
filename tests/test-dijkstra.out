2
20
a1->b1->d1->